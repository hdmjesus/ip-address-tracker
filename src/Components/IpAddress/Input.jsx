import React from "react";
import iconArrrow from "../../Statics/images/icon-arrow.svg";
export const IpAddressInput = () => {
  return (
    <form className="  m-auto flex w-full ">
      <div className="flex min-w-200 justify-between bg-white rounded-lg  h-9 m-auto  w-full   ">
        <input
          className=" w-full p-2 text-xs bg-transparent outline-none "
          type="text"
          placeholder="Search for Any IP Address or Domain"
        />
        <button type="submit" className="bg-black h-9 w-9 rounded-r-lg">
          <img src={iconArrrow} alt="iconArrow" className="w-3 m-auto" />
        </button>
      </div>

      {/* <button type="submit" className="bg-black h-9 w-9">
        <img src={iconArrrow} alt="iconArrow" className="w-3" />
      </button> */}
    </form>
  );
};
