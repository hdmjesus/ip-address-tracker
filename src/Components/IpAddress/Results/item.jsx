import React from "react";

export const IpAddressResultsitem = ({ category }) => {
  return (
    <article className="m-auto mb-4 uppercase font-rubik ">
      <h4 className="text-colorgray font-bold text-xs tracking-wider text-center">
        {category}
      </h4>
      <p className="font-medium text-xl text-blueBlack"> 192.189.98</p>
    </article>
  );
};
