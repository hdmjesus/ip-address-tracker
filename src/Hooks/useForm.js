import { useState } from "react";

export const useForm = (initialState = {}) => {
  const [values, setValues] = useState(initialState);

  const reset = () => {
    setValues(initialState);
  };
  const handleInputChange = ({ target }) => {
    setValues({
      ...values,
      [target.name]: target.value,
    });
  };

  return [values, handleInputChange, reset];
};

// Este Hook nos permite obtener los valores y propiedades de un Input, ademas de tener los valores nuevos cuando este cambie.
