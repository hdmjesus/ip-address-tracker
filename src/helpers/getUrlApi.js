import { apiKey, url } from "../config";

export const getUrlApi = (inputValue) => {
  let UrlApiIp;
  // Esta dondicion , devuelve el endpoint correspondiente a si es una direccion ip o un nombre de dominio
  if (
    inputValue.match(
      /^((25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(25[0-5]|2[0-4]\d|[01]?\d\d?)$/
    )
  ) {
    UrlApiIp = `${url}?apiKey=${apiKey}&ipAddress=${inputValue}`;
    return UrlApiIp;
  } else {
    UrlApiIp = `${url}?apiKey=${apiKey}&domain=${inputValue}`;
    return UrlApiIp;
  }
};
