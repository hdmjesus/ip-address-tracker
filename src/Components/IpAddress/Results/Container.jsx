import React, { useEffect, useState } from "react";
import { IpAddressResultsitem } from "./item";
import { useGlobalData } from "../../../Hooks/useGlobalData";

export const IpAddressResultsContainer = () => {
  const { data, setValueInput } = useGlobalData();
  const [IpAddress, setIpAddress] = useState(data);
  useEffect(() => {
    setIpAddress(data);
    console.log(data);
  }, [data]);

  return (
    <section className="w-11/12 bg-white p-3 rounded-md mt-5 m-auto flex flex-col">
      <IpAddressResultsitem category={"IP Address"} />
      <IpAddressResultsitem category={"Location"} />
      <IpAddressResultsitem category={"Timezone"} />
      <IpAddressResultsitem category={"ips`"} />
    </section>
  );
};
