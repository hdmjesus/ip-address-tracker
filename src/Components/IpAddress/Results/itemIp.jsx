import React, { useEffect } from "react";
import { useGlobalData } from "../../../Hooks/useGlobalData";

export const IpAddressResultsitemIp = ({ category, ip }) => {
  const { loading } = useGlobalData();
  return (
    <article className="m-auto mb-4 uppercase font-rubik ">
      <h4 className="text-colorgray font-bold text-xs tracking-wider text-center">
        {category}
      </h4>
      {!loading && <p className="font-medium text-xl text-blueBlack"> {ip}</p>}
    </article>
  );
};
