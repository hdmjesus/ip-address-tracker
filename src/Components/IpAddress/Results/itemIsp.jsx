import React, { useEffect } from "react";
import { useGlobalData } from "../../../Hooks/useGlobalData";

export const IpAddressResultsitemIsp = ({ category, isp }) => {
  const { loading } = useGlobalData();
  return (
    <article className="m-auto mb-4 uppercase font-rubik ">
      <h4 className="text-colorgray font-bold text-xs tracking-wider text-center">
        {category}
      </h4>
      {!loading && <p className="font-medium text-xl text-blueBlack"> {isp}</p>}
    </article>
  );
};
