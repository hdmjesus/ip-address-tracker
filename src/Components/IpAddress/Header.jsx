import React from "react";
import { IpAddressInput } from "./Input";
import { IpAddressResultsContainer } from "./Results/Container";
import bgHeader from "../../Statics/images/pattern-bg.png";

export const IpAddressHeader = () => {
  return (
    <header
      className="w-full h-80 p-7  m-auto z-10"
      style={{
        backgroundImage: `url(${bgHeader})`,
        backgroundRepeat: "no-repeat",
      }}
    >
      <h1 className="  font-rubik font-bold text-white text-xl text-center mb-4">
        IP Address Tracker
      </h1>
      <IpAddressInput />
      <IpAddressResultsContainer />
    </header>
  );
};
