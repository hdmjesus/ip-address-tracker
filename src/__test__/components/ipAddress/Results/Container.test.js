import React from "react";
import { mount } from "enzyme";
import { TextContext } from "../../../../Context/StaticContext";
import { IpAddressResultsContainer } from "../../../../Components/IpAddress/Results/Container";
import { IpAddressInput } from "../../../../Components/IpAddress/Input";
import { renderHook, act } from "@testing-library/react-hooks";
import { useFetch } from "../../../../Hooks/useFetch";
import { getUrlApi } from "../../../../helpers/getUrlApi";
describe("pruebas en <IpAddressResultsContainer/>", () => {
  const setValueInput = jest.fn();
  const { Provider } = TextContext;
  let loading = true;
  let data = null;
  const wrapper = mount(
    <Provider value={{ data, loading, setValueInput }}>
      <IpAddressResultsContainer />
    </Provider>
  );
  test("should show component when loading is true ", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar la direccion ip correctamente", async () => {
    const { result, waitForNextUpdate } = renderHook(() =>
      useFetch(getUrlApi("8.8.8.8"))
    );

    await waitForNextUpdate({ timeout: 2000 });
    const { data, loading, error } = result.current;
    const wrapper = mount(
      <Provider value={{ data, loading, setValueInput }}>
        <IpAddressInput />
        <IpAddressResultsContainer />
      </Provider>
    );

    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });

    expect(
      wrapper.find("IpAddressResultsitemIp").find("p").text().trim()
    ).toEqual(data.ip);
  });

  test("debe mostrar la direccion geografica correctamente", async () => {
    const { result, waitForNextUpdate } = renderHook(() =>
      useFetch(getUrlApi("8.8.8.8"))
    );

    await waitForNextUpdate({ timeout: 2000 });
    const { data, loading, error } = result.current;
    const wrapper = mount(
      <Provider value={{ data, loading, setValueInput }}>
        <IpAddressInput />
        <IpAddressResultsContainer />
      </Provider>
    );

    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });

    expect(
      wrapper.find("IpAddressResultsitemLocation").find("p").text().trim()
    ).toBe(`${data.location.city}, ${data.location.postalCode}`);
  });

  test("debe mostrar el ISP", async () => {
    const { result, waitForNextUpdate } = renderHook(() =>
      useFetch(getUrlApi("8.8.8.8"))
    );

    await waitForNextUpdate({ timeout: 2000 });
    const { data, loading, error } = result.current;
    const wrapper = mount(
      <Provider value={{ data, loading, setValueInput }}>
        <IpAddressInput />
        <IpAddressResultsContainer />
      </Provider>
    );

    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });

    expect(
      wrapper.find("IpAddressResultsitemIsp").find("p").text().trim()
    ).toBe(data.isp);
  });
});
