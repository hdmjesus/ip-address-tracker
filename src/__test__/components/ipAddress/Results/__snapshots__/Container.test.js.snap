// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <IpAddressResultsContainer/> should show component when loading is true  1`] = `
<section
  className="w-11/12 bg-white p-3 rounded-md mt-5 m-auto flex flex-col"
>
  <article
    className="m-auto mb-4 uppercase font-rubik "
  >
    <h4
      className="text-colorgray font-bold text-xs tracking-wider text-center"
    >
      IP Address
    </h4>
  </article>
  <article
    className="m-auto mb-4 uppercase font-rubik "
  >
    <h4
      className="text-colorgray font-bold text-xs tracking-wider text-center"
    >
      Location
    </h4>
  </article>
  <article
    className="m-auto mb-4 uppercase font-rubik "
  >
    <h4
      className="text-colorgray font-bold text-xs tracking-wider text-center"
    >
      ISP
    </h4>
  </article>
</section>
`;
